sdk/eventhub/azure-eventhub/azure/eventhub/_utils.py

```diff
@@ -135,21 +166,20 @@ def trace_message(event, parent_span=None):
             current_span = parent_span or span_impl_type(
                 span_impl_type.get_current_span()
             )
-            message_span = current_span.span(name="Azure.EventHubs.message")
-            message_span.start()
-            app_prop = dict(event.properties) if event.properties else dict()
-            app_prop.setdefault(
-                b"Diagnostic-Id", message_span.get_trace_parent().encode("ascii")
-            )
-            event.properties = app_prop
-            message_span.finish()
+            with current_span.span(name="Azure.EventHubs.message") as message_span:
+                message_span.kind = SpanKind.PRODUCER
+                app_prop = dict(event.properties) if event.properties else dict()
```

Comment:
What type is `event.properties`? It seems to me like copying all the properties could be expensive...

Reply:
That's a good question, this one was done a while back, I think I got some trouble because by default it was not writable, as a wrapper on some uAMQP things (i.e. `event.properties['foo'] = 42` was denied).
I need to double-check if I can write into it now.

Reply:
It's type of dictionary. Should be able to update it as long as it's not None.
`event.properties = dict(), event.properties['foo'] = 42`

Reply:
For trace: app properties is type:
https://github.com/Azure/azure-uamqp-python/blob/813e3ce6636016948f32d5afa8435c8bfe39b0f2/src/annotations.pyx#L166

(still reading)

Reply:
Since you assign back app_prop to event.properties, you may be able to update event.properties directly?

Reply:
Fixed